packages/svelte/src/internal/client/runtime.js

```diff
@@ -744,12 +734,9 @@ export function get(signal) {
 			current_dependencies_index++;
 		} else if (current_dependencies === null) {
 			current_dependencies = [signal];
-		} else if (signal.read !== current_consumer_read_clock) {
+		} else if (signal !== current_dependencies.at(-1)) {
```

Comment:
isn't `at` like really slow compared to [.length - 1]?

Reply:
I thought it was faster?

Reply:
https://www.measurethat.net/Benchmarks/Show/17464/0/arrayarraylength---1-vs-arrayat-1

Reply:
Hmm, .at is faster on that site, but I got 2.5s for .at and 670ms for .length - 1, when running this snippet in my console,

```js
{
	let N = 1000000;
	const arr = Array.from({length: 100}).map(Math.random);
	let acc= 0;
	console.log(arr.at(-1), arr[arr.length - 1])

	for (var j = 1000; j--;) acc += arr.at(-1);
	for (var j = 1000; j--;) acc += arr[arr.length - 1];
	console.time();
	for (var i = N; i--;) { acc = 0; for (var j = 1000; j--;) acc += arr.at(-1); }
	console.timeEnd();

	console.time();
	for (var i = N; i--;) { acc = 0; for (var j = 1000; j--;) acc += arr[arr.length - 1]; }
	console.timeEnd();

    console.log(acc);
}
```

I read about it being slower somewhere because the method had to also consider `ArrayLike` objects

Reply:
ESBench results with the same code from measurethat
![image](https://github.com/sveltejs/svelte/assets/50981692/36f49c11-6feb-41b0-8c8c-3132514e5ff8)

Reply:
I'll put that change in the other PR.

Reply:
![image](https://github.com/sveltejs/svelte/assets/132045/68044850-ee62-4e84-97dc-e39ac2744bec)

On the left side is `at()` function, and on the right side is `length()` function.

```js
function at(i) {
  return i.at(-1);
}
function length(i) {
  return i[i.length - 1]
}
```

Reply:
Looks like the left side is slightly better.

Reply:
I think that `length()` function should be faster, it will execute 0-6e instructions if nothing goes wrong like OOB, wrong object shape, etc. And `at()` function will execute 0-5d, jmp to 84-8c, jmp to 6a-78. But I am not 100% sure about it, just quickly looked at it :)